broadinstitute/single_cell_portal_core
# Enabling CI via GitHub Actions, self-hosted runner (SCP-3934)

Since our Travis project has been irrevocably broken for weeks now, we have been operating without functional continuous integration in the `single_cell_portal_core` repository.  This update adds the same CI coverage through GitHub Actions, and a private, self-hosted runner that has permission to communicate with our staging database via a [custom Google networking firewall route](https://console.cloud.google.com/networking/firewalls/details/broad-singlecellportal-gh-runner-allow-mongo?project=broad-singlecellportal-staging).

MANUAL TESTING
1. Pull this branch and create a new one based off of this branch
2. Add that new branch name after line 6 in `./github/workflows/run-all-tests.sh`
3. Push the commit, then visit the [GitHub Actions dashboard](https://github.com/broadinstitute/single_cell_portal_core/actions) to find the instance of our CI pipeline running based off of your commit (note: you do not need to let it run to completion, you can cancel the workflow after it has spawned).

Alternatively, you can look below to see that an instance of `SCP Continuous Integration / Run-All-Tests (pull_request)` has spawned in this pull request, as well as post-merge commits to `development` and `master`.

A history of all previous runs/workflows can be found [here](https://github.com/broadinstitute/single_cell_portal_core/actions).

This PR satisfies SCP-3934.
